src/CachePlugin.php

```diff
@@ -87,8 +114,17 @@ public function handleRequest(RequestInterface $request, callable $next, callabl
                     $response = $response->withBody($this->streamFactory->createStream($body));
                 }
 
-                $cacheItem->set(['response' => $response, 'body' => $body])
-                    ->expiresAfter($this->getMaxAge($response));
+                $maxAge = $this->getMaxAge($response);
+                $currentTime = time();
+                $cacheItem
+                    ->expiresAfter($this->config['cache_lifetime'] + $maxAge)
```

Comment:
when doing cache validation, we keep outdated caches and send If-Modified-Since / If-None-Match requests to the server to check if the cache is ok to be used. we might add some comment in the code though, to make it clear.